src/ripple/protocol/Serializer.h

```diff
@@ -280,6 +282,20 @@ class Serializer
     int addEncoded (int length);
 };
 
+template<class FwdIt>
+int Serializer::addVL(FwdIt begin, FwdIt end, int len)
+{
+    int ret = addEncoded(len);
+    if (len)
+        for (; begin != end; ++begin)
```

Comment:
1. I used `len` because I based this off of the `addVL(const void* ptr, int len)` override at https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/impl/Serializer.cpp#L213-L221. On further thought, since the loop checks `begin != end`, there really is no need for any separate check.
2. `FwdIt` wasn't the best name for the template param. Would `Iter` be better? It really doesn't tell us anything about the iterator, or the underlying object anyway. It just has to be something that exposes `data()` and `size()`.
3. I don't like the idea of calculating `len` like that, though, because the iterators could easily be pointing to differently sized data that simply isn't guaranteed to be contiguous. For example, an array of differently-sized `std::string`s. If we want to compute `len`, the only _safe_ way to do it is to make two passes over the iterators. Doable, but obviously slower.